File: reduks-logger/src/main/kotlin/com/beyondeye/reduks/logger/logformatter/LogFormatter.java
package com.beyondeye.reduks.logger.logformatter;

/**
 * LogFormatter is a wrapper of {@link android.util.Log}
 * But more pretty, simple and powerful
 * adapted from https://github.com/orhanobut/logger
 */
public class LogFormatter {
//  public static final int DEBUG = 3;
//  public static final int ERROR = 6;
//  public static final int ASSERT = 7;
//  public static final int INFO = 4;
//  public static final int VERBOSE = 2;
//  public static final int WARN = 5;

  private LoggerPrinter printer = new LoggerPrinter();

  public LogFormatter(String tag) {
    printer = new LoggerPrinter();
    printer.init(tag);
  }
  public void resetSettings() {
    printer.resetSettings();
  }
  public LogFormatterSettings getSettings() { return printer.getSettings(); }

  public  LoggerPrinter t(String tag) {
    return printer.t(tag, printer.getSettings().getMethodCount());
  }

  public  LoggerPrinter t(int methodCount) {
    return printer.t(null, methodCount);
  }

  public  LoggerPrinter t(String tag, int methodCount) {
    return printer.t(tag, methodCount);
  }

  public  void log(int priority, String tag, String message, Throwable throwable) {
    printer.log(priority, tag, message, throwable);
  }

//  public static void d(String message, Object... args) {
//    printer.d(message, args);
//  }

//  public static void d(Object object) {
//    printer.d(object);
//  }

//  public static void e(String message, Object... args) {
//    printer.e(null, message, args);
//  }

//  public static void e(Throwable throwable, String message, Object... args) {
//    printer.e(throwable, message, args);
//  }

//  public static void i(String message, Object... args) {
//    printer.i(message, args);
//  }
//
//  public static void v(String message, Object... args) {
//    printer.v(message, args);
//  }
//
//  public static void w(String message, Object... args) {
//    printer.w(message, args);
//  }
//
//  public static void wtf(String message, Object... args) {
//    printer.wtf(message, args);
//  }

  /**
   * Formats the json content and print it
   *
   * @param json the json content
   */
  public  void json(String json) {
    printer.json(json);
  }



}

File: reduks-logger/src/main/kotlin/com/beyondeye/reduks/logger/logformatter/LogFormatter.kt
package com.beyondeye.reduks.logger.logformatter

import com.beyondeye.reduks.logger.LogLevel

/**
 * LogFormatter is a wrapper of android.util.Log
 * But more pretty, simple and powerful
 * adapted from https://github.com/orhanobut/logger
 * It also support log entry grouping i.e. common indenting of multiple subsequent lines in a similar way to javascript console:
 * see https://developer.mozilla.org/en/docs/Web/API/console (Using groups in console)
 */
class LogFormatter(tag: String, settings: LogFormatterSettings) {

    private val printer: LogFormatterPrinter

    init {
        printer = LogFormatterPrinter(settings)
        printer.setTag(tag)
    }

    fun resetSettings() {
        printer.resetSettings()
    }

    val settings: LogFormatterSettings
        get() = printer.settings


    fun setLocalMethodCount(methodCount: Int) {
         printer.selLocalMethodCount(methodCount)
    }

    /**
     * start a new group, increasing indent, until next call to [groupEnd]
     * and call [log] with the specified argument
     */
    fun group(groupHeaderMessage: String, logLevel: Int= LogLevel.INFO,tagSuffix: String?=null) {
        printer.groupStart()
        printer.log(groupHeaderMessage,logLevel, tagSuffix) //group header
        printer.increaseIndent() //increase indent AFTER header
    }

    /**
     * start a new collapsed group: all log lines are collapsed to a single line,, until next call to [groupEnd]
     * and call [log] with the specified argument
     */
    fun groupCollapsed(groupHeaderMessage: String, logLevel: Int=LogLevel.INFO,tagSuffix: String?=null) {
        printer.groupCollapsedStart()
        printer.log(groupHeaderMessage,logLevel, tagSuffix)
        printer.increaseIndent() //increase indent AFTER header
    }

    fun groupEnd() {
        printer.groupEnd()
    }
    fun log(message: String, logLevel: Int=LogLevel.INFO,tagSuffix: String?=null,throwable: Throwable?=null) {
        val message_=printer.addFormattedThrowableToMessage(message,throwable)
        printer.log(message_,logLevel, tagSuffix)
    }

    /**
     * Formats the json content and print it

     * @param jsonMessage the json content
     */
    fun json(objName: String, jsonMessage: String, logLevel: Int=LogLevel.INFO,tagSuffix: String?=null) {
        printer.json(objName, jsonMessage,logLevel, tagSuffix)
    }
    fun getPrettyPrintedJson(objName: String, jsonMessage: String):String {
        return printer.getPrettyPrintedJson(objName,jsonMessage)
    }

    fun getLogAsString(): String  {
        val logAdapter = printer.settings.logAdapter
        if (logAdapter !is StringBufferLogAdapter) return ""
        return logAdapter.buffer
    }

}
